File: packages/typescript/src/resources/capabilities.ts
import { capabilitiesGet } from "../_internal/api/sdk.gen.js";
import type { CapabilitiesOutputBody } from "../_internal/api/types.gen.js";
import type { ResourceClient } from "./_client.js";

export type { CapabilitiesOutputBody };

/** Server-advertised capabilities: version, feature flags, license, scopes. */
export class CapabilitiesResource {
  constructor(private readonly client: ResourceClient) {}

  /**
   * Fetch the capabilities the server exposes to the current key. `GET /capabilities`
   *
   * @returns The capabilities document, or `null` on fail-open error.
   */
  async get(): Promise<CapabilitiesOutputBody | null> {
    return this.client.invoke(capabilitiesGet, {});
  }
}

File: packages/python/src/axonpush/resources/capabilities.py
"""Capabilities resource — server feature flags, scopes and license. ``/capabilities``."""

from __future__ import annotations

from typing import TYPE_CHECKING

from axonpush._internal.api.api.capabilities import capabilities_get as _get_op
from axonpush._internal.api.models import CapabilitiesOutputBody

if TYPE_CHECKING:
    from axonpush.resources._base import AsyncClientProtocol, SyncClientProtocol


class Capabilities:
    """Server-advertised capabilities for the calling key."""

    def __init__(self, client: SyncClientProtocol) -> None:
        self._client = client

    def get(self) -> CapabilitiesOutputBody | None:
        """Fetch capabilities (version, feature flags, scopes, license). ``GET /capabilities``"""
        return self._client._invoke(_get_op)


class AsyncCapabilities:
    """Async sibling of :class:`Capabilities`."""

    def __init__(self, client: AsyncClientProtocol) -> None:
        self._client = client

    async def get(self) -> CapabilitiesOutputBody | None:
        """See :meth:`Capabilities.get`."""
        return await self._client._invoke(_get_op)
